### ISSUE TITLE
Fix for ValueError: continuous format is not supported

### ISSUE THREAD

**Issue by nicozhou7** [rvorias/ind_knn_ad#7]
in the indad/model.py line 78

the mask must be convert to int type, otherwise, it will get **ValueError: continuous format is not supported**

Here is the solution: I changed the mask type from float to int 

`pixel_labels.extend(mask.flatten().numpy().astype(int))`

The initial function is like below 

```
	def evaluate(self, test_ds: VisionDataset) -> Tuple[float, float]:
		"""Calls predict step for each test sample."""
		image_preds = []
		image_labels = []
		pixel_preds = []
		pixel_labels = []

		for sample, mask, label in tqdm(test_ds, **TQDM_PARAMS):
			z_score, fmap = self.predict(sample.unsqueeze(0))
			
			image_preds.append(z_score.numpy())
			image_labels.append(label)
			
			pixel_preds.extend(fmap.flatten().numpy())
			pixel_labels.extend(mask.flatten().numpy())
			
		image_preds = np.stack(image_preds)

		image_rocauc = roc_auc_score(image_labels, image_preds)

		pixel_rocauc = roc_auc_score(pixel_labels, pixel_preds)

		return image_rocauc, pixel_rocauc
```

**Comment by rvorias** [comment]
I've never encountered this issue. Do you have a command that can reproduce this?

Are you maybe using custom data?

**Comment by nicozhou7** [comment]
Thanks for the reply, this happen for three of them, and I am not using custom data

`python indad/run.py spade --dataset transistor_reduced`
`python indad/run.py padim --dataset transistor_reduced`
`python indad/run.py pathcore --dataset transistor_reduced`

This is my system specifications:

- Windows 10 
- Anaconda
- Python 3.8
- scikit-learn==0.23.2
- torch==1.8.1

**Comment by rvorias** [comment]
Thanks for the commands! I figured out what was wrong: I didn't use nearest neighbors when resizing the ground truth masks. 
I'll change the masks and see to push your suggestion later this evening.

**Comment by nicozhou7** [comment]
So, if you use nearest neighbors when resizing the GT Mask, then `pixel_labels` does not need to be casted to integer anymore, am I right? 

One more question, i want to visualize the outputs of the segmentation, how can I do that?

**Comment by rvorias** [comment]
Yep. I'd rather set the input data right than cast the labels to int as casting to int has a floor effect: e.g. 0.99 will still become 0.

**Comment by nicozhou7** [comment]
Great, waiting for your fix! and how about the visualization of the predicted segmentation? Should I directly use the `pixel_preds` as the segmentation output? Any post processing needed?

**Comment by rvorias** [comment]
The evaluation function comes in handy as it can give you a threshold for a certain desired recall.
If you want to visualize your predictions, it's easier to call 
```python
# get predictions
img_lvl_anom_score, pxl_lvl_anom_score = model.predict(test_ds[0].unsqueeze(0))
```
Or something like that. 
You can then use the threshold that you got from your aurocs to threshold the faulty pixels. 
Right now, you'll have to extract the best threshold yourself.

**Comment by rvorias** [comment]
FYI, the reduced datasets are not supposed to be run with the evaluate function.
You better choose just "hazelnut" or "transistor" (the data is automatically downloaded)

**Comment by rvorias** [comment]
Fixed in 2cd595c97282cb753cddf64691de1fa8c5b79943
